otherlibs/systhreads/st_stubs.c

```diff
@@ -699,7 +701,7 @@ static void * caml_thread_tick(void * arg)
     atomic_store_release(&domain->requested_external_interrupt, 1);
     caml_interrupt_self();
   }
-  return NULL;
+  return 0;
```

Comment:
I think it would be nice to make all thread routines return an explicitly cast value, i.e.
```suggestion
  return (CAML_THREAD_FUNCTION)0;
```

But maybe that's too pedantic.

Reply:
It's a bit pendantic, but doesn't hurt. I'll add this where needed.

Reply:
Ah, of course this doesn't work as-is because of the `WINAPI` specifier ;)

Reply:
Damn. No casts, then.

Reply:
Note how it is defined:
```c
#ifdef _WIN32
int caml_plat_thread_create(caml_plat_thread *restrict,
                            const caml_plat_thread_attr *restrict,
                            unsigned (WINAPI *)(void *),
                            void *restrict);
#define CAML_THREAD_FUNCTION unsigned WINAPI
#else
/* ... */
#define CAML_THREAD_FUNCTION void *
#endif
```

The `WINAPI` macro needs to be put inside the parentheses for the function pointer type!

Applied to `return (CAML_THREAD_FUNCTION) 0;`, MSVC complains:

```
runtime/domain.c(1095): warning C4229: anachronism used: modifiers on data are ignored
runtime/domain.c(1278): warning C4229: anachronism used: modifiers on data are ignored
```

I think it'd be too heavy introducing a new macro without the `WINAPI` part, and also ugly to do (if I were to remove the macro):

```c
static
#ifdef _WIN32
unsigned WINAPI
#else
void *
#endif
my_thread_func(void *arg) {
  /* ... */
#ifdef _WIN32
  return 0;
#else
  return NULL;
#endif
}
```

With your permission, I won't include this change; unless you have a better idea on how to write it.

As a reminder, `WINAPI` is a macro that evaluates to [`__stdcall`](https://learn.microsoft.com/en-us/cpp/cpp/stdcall?view=msvc-170), a Microsoft-specific modifier specifying the calling convention used to call Win32 API functions. It only has a meaning on x86 and is ignored on other architectures.

Reply:
No, that's fine to rely upon the implicit cast from `int` to `void *`.